rust/arrow/src/array/equal.rs

```diff
@@ -692,6 +692,22 @@ impl ArrayEqual for StructArray {
     }
 }
 
+impl ArrayEqual for UnionArray {
+    fn equals(&self, _other: &dyn Array) -> bool {
+        unimplemented!()
```

Comment:
Is this intentionally unimplemented? If so, would be good to add a comment explaining that.